ops/__init__.py

```diff
@@ -321,3 +321,47 @@
 # rather than a runtime concern.
 
 from .version import version as __version__
+
+
+class _Main:
+    def __call__(
+        self, charm_class: Type[charm.CharmBase], use_juju_for_storage: Optional[bool] = None
+    ):
+        return _main.main(charm_class=charm_class, use_juju_for_storage=use_juju_for_storage)
+
+    def main(
+        self, charm_class: Type[charm.CharmBase], use_juju_for_storage: Optional[bool] = None
+    ):
+        return _legacy_main.main(
+            charm_class=charm_class, use_juju_for_storage=use_juju_for_storage
+        )
+
+
+main = _Main()
+"""Set up the charm and dispatch the observed event.
+
+Recommended usage:
+
+.. code-block:: python
+
+    import ops
+
+
+    class SomeCharm(ops.CharmBase):
+        ...
+
+
+    if __name__ == "__main__":
+        ops.main(SomeCharm)
```

Comment:
Nit: Can we remove the double blank lines? I realise we'd do this in real code (formatter might even require it), but it feels like a lot of vertical whitespace in the docs. Also, let's see if I can sneak in a change from `"` to `'`. :-)

```suggestion
    import ops

    class SomeCharm(ops.CharmBase):
        ...

    if __name__ == '__main__':
        ops.main(SomeCharm)
```